UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Pipenets/Pipe.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Atmospherics;
+
+public class Pipe : PickUpTrigger
+{
+	public List<Pipe> nodes = new List<Pipe>();
+	public bool anchored = false;
+	public Direction direction = Direction.NORTH;
+	public Sprite[] pipeSprites;
+	public SpriteRenderer spriteRenderer;
+
+	public Pipenet pipenet;
+	public float volume = 70;
+
+	public int ARANpipenetmembers;
+	public string ARANpipenetName;
+	public float ARANpipenetVolume;
+
+	public enum Direction
+	{
+		NORTH,
+		SOUTH,
+		WEST,
+		EAST
+	}
+
+	public override bool Interact(GameObject originator, Vector3 position, string hand)
+	{
+		if (!CanUse(originator, hand, position, false))
+		{
+			return false;
+		}
+		if (!isServer)
+		{
+			//ask server to perform the interaction
+			InteractMessage.Send(gameObject, position, hand);
+			return true;
+		}
+
+		PlayerNetworkActions pna = originator.GetComponent<PlayerNetworkActions>();
+		GameObject handObj = pna.Inventory[hand].Item;
+
+		if(handObj == null)
+		{
+			if(!anchored){
+				return base.Interact(originator, position, hand);
+			}
+		}
+		else{
+			if (handObj.GetComponent<WrenchTrigger>())
+			{
+				if (anchored)
+				{
+					anchored = false;
+					Detach();
+				}
+				else
+				{
+					if (GetAnchoredPipe(transform.position) != null)
+					{
+						return true;
+					}
+					CalculateAttachedNodes();
+					Attach();
+					anchored = true;
+				}
+				SpriteChange();
+				SoundManager.PlayAtPosition("Wrench", transform.position);
+			}
+		}
+
+		return true;
+	}
+
+
+	public virtual void Attach()
+	{
+		Pipenet foundPipenet = null;
+		for (int i = 0; i < nodes.Count; i++)
+		{
+			foundPipenet = nodes[i].pipenet;
+			break;
+		}
+		if (foundPipenet == null)
+		{
+			foundPipenet = new Pipenet();
+		}
+		foundPipenet.AddPipe(this);
+
+		transform.rotation = new Quaternion();
+		transform.position = Vector3Int.RoundToInt(transform.position);
```

Comment:
Do they still get blown away by wind? You could take advantage of `IsTileSnap` and `isNotPushable` bools, making sure they can't be blown away or nudged (and snap to tile like lockers and players)